common/functions/src/scalars/logics/logic.rs

```diff
@@ -53,7 +53,8 @@ impl Function for LogicFunction {
         Ok(DataType::Boolean)
     }
 
-    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+    // Null is treated as false, so it should not output any Null.
+    fn nullable(&self, _arg_fields: &[DataField]) -> Result<bool> {
```

Comment:
select  null and 3;
select not null;
select null or 3;

all returns null

Reply:
OK, will fix. But if that is how MySQL handles logic function, do we really want to be  the same ? 
Especially ``` select not null; -> null``` and ``` select null or 3; -> null ```  it is confusing ...

Reply:
Find some useful doc from postgre's doc "SQL uses a three-valued logic system with true, false, and null, which represents "unknown". "

Reply:
> select null and 3; select not null; select null or 3;
> 
> all return null

`select null or 3 ` will return `1`(means true).

> OK, will fix. But if that is how MySQL handles logic function, do we really want to be the same ?
> Especially ` select not null; -> null` and `select null or 3; -> null` it is confusing ...

The return result of `select not null` in clickhouse is also `null`.

Reply:
> > select null and 3; select not null; select null or 3;
> > all return null
> 
> Correct: `select null or 3 ` will return `1`(means true).
> 
> > OK, will fix. But if that is how MySQL handles logic function, do we really want to be the same ?
> > Especially ` select not null; -> null` and `select null or 3; -> null` it is confusing ...
> 
> The return result of `select not null` in clickhouse is also `null`. So we can just use the default.

Right, I found this doc explains well https://www.postgresql.org/docs/9.1/functions-logical.html